agents/agents-core/src/commonTest/kotlin/ai/koog/agents/core/dsl/extension/ParallelNodesTest.kt

```diff
@@ -0,0 +1,107 @@
+package ai.koog.agents.core.dsl.extension
+
+import ai.koog.agents.core.agent.AIAgent
+import ai.koog.agents.core.agent.config.AIAgentConfig
+import ai.koog.agents.core.agent.entity.AIAgentStorageKey
+import ai.koog.agents.core.dsl.builder.forwardTo
+import ai.koog.agents.core.dsl.builder.strategy
+import ai.koog.agents.core.tools.ToolRegistry
+import ai.koog.agents.features.eventHandler.feature.EventHandler
+import ai.koog.agents.testing.tools.DummyTool
+import ai.koog.agents.testing.tools.getMockExecutor
+import ai.koog.agents.testing.tools.mockLLMAnswer
+import ai.koog.prompt.dsl.prompt
+import ai.koog.prompt.llm.OllamaModels
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class ParallelNodesTest {
+
+    @Test
+    fun testContextSubstitution() = runTest {
+        // Create a key to store and retrieve values from the context
+        val testKey = AIAgentStorageKey<String>("testKey")
+
+        val agentStrategy = strategy("test-context") {
+            // Create three nodes that modify the context
+            val node1 by node<Unit, String>("node1") {
+                storage.set(testKey, "value1")
+                "Result from node1"
+            }
+
+            val node2 by node<Unit, String>("node2") {
+                storage.set(testKey, "value2")
+                "Result from node2"
+            }
+
+            val node3 by node<Unit, String>("node3") {
+                storage.set(testKey, "value3")
+                "Result from node3"
+            }
+
+            // Create a parallel node that executes all three nodes
+            val parallelNode by parallel<Unit, String>(
+                node1, node2, node3,
+                name = "parallelNode",
+            )
+
+            val reduceNode by reduce<Unit, String>(name = "reduceNode") { results ->
```

Comment:
parallel + reduce -- is it a good pair of names?

Also, is there a way to not call reduce right after parallel?

Something like parallel(....).transform(...).transform(...).transform(...).reduce(...)

Something similar to:

myList.map { // parallel
   it.toInt()
}.map { // transform
   it + 5
}.map{ // transform 
   println(it)
   it-1
}.max() // reduce

Reply:
Map is taken so maybe fork + reduce?

Reply:
> Map is taken

Sorry, I meant `transform` :)